scripts/monorepo/src/isConvergedPackage.js

```diff
@@ -38,6 +39,47 @@ function isConvergedPackage(options = {}) {
   return semver.major(packageJson.version) >= 9 || isNightlyVersion(packageJson.version) || hasVNextTag;
 }
 
+/**
+ * @param {Object} [options]
+ * @param {PathOrPackageJson} [options.packagePathOrJson] - optional different package path to run in OR previously-read package.json
+ * (defaults to reading package.json from `process.cwd()`)
+ * @returns
+ */
+function shipsAMD(options = {}) {
+  const { packagePathOrJson } = options;
+  const packageJson =
+    !packagePathOrJson || typeof packagePathOrJson === 'string'
+      ? readConfig('package.json', /** @type {string|undefined} */ (packagePathOrJson))
+      : packagePathOrJson;
+
+  if (!packageJson) {
+    throw new Error(`package.json doesn't exist`);
+  }
+
+  const metadata = getProjectMetadata({ name: packageJson.name });
+
+  if (metadata.projectType !== 'library') {
+    return false;
+  }
+
+  const tags = new Set(metadata.tags ?? []);
+  const isV8 = tags.has('v8');
+  const isV9 = tags.has('vNext');
```

Comment:
can we reuse the `isConvergedPackage` check?

Reply:
I wanted to do that initially, but the overhead of re-parsing the same json files with logic was not worth the hassle. also it decouples the logic/domain which is what we should adhere to.  It will be removed in the future as this check should apply to all v9 etc, because importing from  directory is not part of ESM spec either. (under the hood I'd assume compiler will handle that - not the case ATM )